CompareRaftMembers/DiffStreamOplogFilter.js

```diff
@@ -0,0 +1,240 @@
+const async = require('async');
+const http = require('http');
+const stream = require('stream');
+
+const RaftOplogStream = require('./RaftOplogStream');
+
+const DEFAULT_MAX_BUFFERED_ENTRIES = 10000;
+const DEFAULT_RETRY_DELAY_MS = 1000;
+const DEFAULT_MAX_RETRY_DELAY_MS = 10000;
+
+const httpAgent = new http.Agent({
+    keepAlive: true,
+});
+
+class DiffStreamOplogFilter extends stream.Transform {
+    constructor(params) {
+        super({ objectMode: true });
+        const {
+            bucketdHost,
+            bucketdPort,
+            maxBufferedEntries: userMaxBufferedEntries,
+            retryDelayMs: userRetryDelayMs,
+            maxRetryDelayMs: userMaxRetryDelayMs,
+        } = params;
+        this.bucketdHost = bucketdHost;
+        this.bucketdPort = bucketdPort;
+        this.maxBufferedEntries = userMaxBufferedEntries || DEFAULT_MAX_BUFFERED_ENTRIES;
+        const retryDelayMs = userRetryDelayMs || DEFAULT_RETRY_DELAY_MS;
+        const maxRetryDelayMs = userMaxRetryDelayMs || DEFAULT_MAX_RETRY_DELAY_MS;
+        this.retryParams = {
+            times: 20,
+            interval: retryCount => Math.min(
+                // the first retry comes as "retryCount=2", hence substract 2
+                retryDelayMs * (2 ** (retryCount - 2)),
+                maxRetryDelayMs,
+            ),
+        };
+        this.bucketNameToRaftSessionId = {};
+        this._initRaftOplogStreams(params);
+
+        this.nBufferedEntries = 0;
+        this.pendingTransformCallback = null;
+        this.pendingFlushCallback = null;
+    }
+
+    _initRaftOplogStreams(params) {
+        const {
+            bucketdHost, bucketdPort,
+            excludeFromCseqs,
+        } = params;
+
+        this.raftSessionStates = {};
+        for (const rsId of Object.keys(excludeFromCseqs)) {
+            const raftSessionState = {};
+            this.raftSessionStates[rsId] = raftSessionState;
```

Comment:
this might be better at the bottom of the block after raftSessionState is initialized with oplogStream, oplogKeys, inputBuffer